server/document.js

```diff
@@ -88,13 +88,13 @@ export class Head extends Component {
     const pagePathname = getPagePathname(pathname)
 
     return <head {...this.props}>
-      {(head || []).map((h, i) => React.cloneElement(h, { key: h.key || i }))}
       {page !== '/_error' && <link rel='preload' href={`${assetPrefix}/_next/${buildId}/page${pagePathname}`} as='script' />}
       <link rel='preload' href={`${assetPrefix}/_next/${buildId}/page/_error.js`} as='script' />
       {this.getPreloadDynamicChunks()}
       {this.getPreloadMainLinks()}
       {styles || null}
       {this.props.children}
+      {(head || []).map((h, i) => React.cloneElement(h, { key: h.key || i }))}
```

Comment:
Is it a good decision to move user's head tags to the bottom of head? Styles are more important for page view than scripts. And some user scripts sometimes need to start working before other scripts. For example ads networks sometimes uses `docuemnt.write` before page loading to inject their scripts. May be `{this.props.children}` bubble to the top?

Reply:
I'm not sure about what it's convenient or not in those cases, can you share an opinion about this @timneutkens  ?

Reply:
I believe this should be the other way around, moving `{this.props.children}` to be the first thing rendered, then head, then the preloads, then styles.

The `preload` tags should come before `{styles}` because preload will trigger an early request for the resource, only after that css parsing happens.